lib/progress.ts

```diff
@@ -84,3 +84,51 @@ export interface IPublishProgress {
    */
   abort(): void;
 }
+
+class GlobalOutputHandler {
+  private progressListener: IPublishProgressListener | undefined;
+  private completionProgress: number;
+
+  constructor(completionProgress: number = 0, progressListener?: IPublishProgressListener) {
+    this.progressListener = progressListener;
+    this.completionProgress = completionProgress;
+  }
+
+  public setListener(listener: IPublishProgressListener) {
+    this.progressListener = listener;
+  }
+
+  public setCompletionProgress(progress: number) {
+    this.completionProgress = progress;
+  }
+
+  public publishEvent(eventType: EventType = EventType.DEBUG, text: string) {
+    const progressEvent: IPublishProgress = {
+      message: text,
+      abort: () => {},
+      percentComplete: this.completionProgress,
+    };
+    // if no listener is passed we just swallow everything.
+    if (this.progressListener) {
+      this.progressListener.onPublishEvent(eventType, progressEvent);
+    }
+  }
+
+  public verbose(text: string) {
+    this.publishEvent(EventType.DEBUG, text);
+  }
+
+  public error(text: string) {
+    this.publishEvent(EventType.FAIL, text);
+  }
+
+  public info(text: string) {
+    this.publishEvent(EventType.SUCCESS, text);
```

Comment:
Mapping `info` to `EventType.SUCCESS` does seem right. `SUCCESS` indicates that asset publishing has successfully finished, at least according to the doc string:

https://github.com/cdklabs/cdk-assets/blob/cacbcf4a7fb8af3107e43331c5f9c3bb95f497b9/lib/progress.ts#L22-L25

It might be the code currently doesn't make assumption on the event type being emitted, but thats likely to change and at some point calls to `info` will stop making sense. Also I see this is only used in `shell.ts` once, does it really merit a public method for it?